Pm 2.0 ga2 kim edits (#66)
mulesoft/docs-partner-manager: modules/ROOT/pages/configure-message-flows.adoc

> = Configure a Message Flow

Reviewer: Create and Configure?

Author: Yes, I think that is best.

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> You can create and configure a message flow at the time you create a trading partner or afterwards. Once you deploy the message flow you configured, it's best to test it.

Reviewer: minimalism
```suggestion
You can create and configure a message flow either when you create a trading partner or afterward. After you deploy the message flow, it's best to test it.
```

* xref:endpoint-https-send.adoc[HTTP and HTTPS Send Configuration]

> . After you create the source, map, and target sections in the interface, confirm that each shows a green check mark next to it in the flow, and click *Deploy* to deploy the configurations to your Mule Runtime.

Reviewer: ```suggestion
. After you create the source, map, and target sections in the interface, confirm that each shows a green check mark next to it in the flow, and click *Deploy* to deploy the configurations to Mule.
```

+

> . Your message flow configuration is deployed to your Mule runtime engine and you can start receiving messages from your partner.

Reviewer: This is not a step; it is a followup to the previous step.
```suggestion
Your message flow configuration is deployed to your Mule runtime engine and you can start receiving messages from your partner.
```

+

> . Once a message flow is successfully deployed, you can track its processed messages from the Activity page.

Reviewer: Also not a step but a followup.
```suggestion
. After a message flow is successfully deployed, you can track its processed messages using the Activity page.
```

== Test Message Flows

> There are a couple of reasons why you should send test messages through the message flow after you deploy:

Reviewer: Just to not begin a sentence with "there are" as subject...:
```suggestion
You should send test messages through the message flow after you deploy to verify the following:
```

> * Ensure that you are receiving messages and any acknowledgments from your partners.

Reviewer: ```suggestion
* You are receiving messages and acknowledgments from your partners.
```

> * Make sure that the DataWeave map is working and your backend systems receive the transformed message.

Reviewer: ```suggestion
* The DataWeave map is working as intended and your backend systems transformed messages.
```

Author: WE need recieved in there because the backend systems don't transform. Gonna leave as is.

> You can test the flow by deploying it in a sandbox environment, then undeploying it, and finally deploying it to a production environment.

Reviewer: ```suggestion
You test the flow by deploying it in a sandbox environment. When you verify that the flow is successful, you then undeploy it from the sandbox and deploy it to your production environment.
```

> Test your message flow in your environment:

Reviewer: Clarify: is this the sandbox text or a new text in "your" (production) environment?

Author: At one point, I had put in there that you can test in either environment. Many apps like to use the sandbox for testing, but some users will only test in production. A reviewer wanted the environments out because he said it was confusing. I will check with the PM for clarity.

Author: Adding the both option back in.

. Initiate a transmission on the receiving endpoint.

> . From within Anypoint Partner Manager select *Activity* and select the transmission from the list.

Reviewer: ```suggestion
. From within Anypoint Partner Manager, select *Activity* and select the transmission from the list.
```

> . Scan the Transmission Flow sections for an exclamation point (!) in any of the rectangle boxes next to the displayed flow stages as this indicates there is an issue with the transmission.

Reviewer: ```suggestion
. Scan for an exclamation point (!) in any of the boxes next to the displayed flow stages, indicating an issue with the transmission.
```

Reviewer: This task might benefit from some screen captures.

> . If view payloads are enabled, review the received payload by clicking *View payload* next to the Received transmission section.

Reviewer: The previous step indicated the possibility of issues with the transmission; does this step address that?

Reviewer: I'm not sure that "view payloads are enabled" is grammatical: a type of payload, or a control called View Payloads?
Clarify...or consider this:
```suggestion
. Review the received payload by clicking *View payload*.
```
Review it for what?
Also, do we need to add a subordinate para that explains what you mean by "if view payloads are enabled"? Are we saying that the controls in this step might not appear in the UI if this has not been enabled?

Author: Fixed.

> . If you had any TA1 and functional acknowledgments (999 or 997) set, review and confirm that they are there.

Reviewer: Just checking: is this step part of the previous one (a substep)?
I ask because the previous step does not specify WHAT to check for, and thought this might be it???

If not...how does one confirm the presence of the TA1 acks? Or is that obvious to your audience?

Author: I will reword. The "if" statements were to indicate that these choices are optional - depending on your set up. Also, they would know what to look for if they want to download and view payloads. The problem is, that the Sys Admin may or may not have set it up that they can view and download the payloads. I think it's kind of a problem with the design. I added some text, but think I could add a bit more. Yes, the user would know if they should be getting TA1 acknowledgments. The would have set them up when they configured their message flow. It's used in EDI and it indicates the status of the processing.

> . If you had any TA1 and functional acknowledgments (999 or 997) set, review and confirm that they are there.

Reviewer: ```suggestion
. If you have TA1 and functional acknowledgments (999 or 997) set, review and confirm that they are there.
```
Review for what?
Where is "there"?

> . Check that the transmission routed messages to the correct message flows.

Reviewer: ```suggestion
. Verify that the transmission routed messages to the correct message flows.
```

> . Ensure that the messages are routed to the target system. Specifically check:

Reviewer: ```suggestion
. Verify that the messages are routed to the target system:
```

> ** That it was successfully delivered to the target endpoint.

Reviewer: ```suggestion
** Successfully delivered to the target endpoint
```

> ** That the message was transformed correctly.

Reviewer: ```suggestion
** Transformed correctly
```
